### ISSUE TITLE
Pressing escape takes a bit to do an action

### ISSUE THREAD

**Issue by conradob** [LukeSmithxyz/voidrice#800]
Sorry to raise an issue just to ask this but when I press escape it takes few ms to take an action.

The most common case is when leaving from insert mode, need to always press escape and wait for it.

Just asking this here because it happens only with DWM, opening a ttyN it works properly with fast  response and I don't know where I can find this config.

**Comment by vladdoster** [comment]
It's because escape is mapped to caps lock. 

The config is found [here](https://github.com/LukeSmithxyz/dwm/blob/master/config.h) but that the remap is done [here](https://github.com/LukeSmithxyz/voidrice/blob/master/.local/bin/remaps)

**Comment by conradob** [comment]
The remap works properly and even if I change escape back to the default behaviour it still take few ms to do the action, in the case, exit insert mode.

**Comment by LukeSmithxyz** [comment]
It sounds like you have a macro/shortcut bound to <escape> plus
something else in vim, perhaps in one of your plugins. In that case, it
is only a visual delay in chaning the cursor pending the next key and
will not actually delay your next keypress in normal mode.

**Comment by conradob** [comment]
Thanks for your support. 

This was also happening when I wasn't in vim, in the zsh with vi-mode also.

Just found that this behaviour was happening only when working with tmux, so I commented all the `tmux.conf` but was still getting the delay when changing the mode.

Investigating it more I found this issue https://github.com/neovim/neovim/issues/2035#issuecomment-75378039 suggesting to add `set -sg escape-time 20` in `tmux.conf` and now I'm getting an instant action when pressing escape.

For more reference related to this problem, see this section in Neovim FAQ 
https://github.com/neovim/neovim/wiki/FAQ#esc-in-tmux-or-gnu-screen-is-delayed
